crawler.go

```diff
@@ -122,10 +142,22 @@ func (c *Crawler) EnqueueWithSource(method string, URL string, sourceURL string)
 		if err != nil {
 			return ok, err
 		}
-		cmd := Cmd{&fetchbot.Cmd{U: u, M: "GET"}, s}
-		err = c.q.Send(cmd)
-		return ok, err
+
+		if c.BasicAuthUser != "" && c.BasicAuthPass != "" {
+			cmd := &CmdBasicAuth{&fetchbot.Cmd{U: u, M: method}, s, c.BasicAuthUser, c.BasicAuthPass}
+
+			if err := c.q.Send(cmd); err != nil {
+				return ok, err
+			}
+		} else {
+			cmd := &Cmd{&fetchbot.Cmd{U: u, M: method}, s}
```

Comment:
And I think you'll have to define `cmd` before entering `if else`, like this `var cmd fetchbot.Command` (because `fetchbot.Send` accepts a `fetchbt.Command` parameter).